arraylist/src/main/java/ua/hillel/ArrayList.java

```diff
@@ -1,68 +1,125 @@
 package ua.hillel;
 
+import java.util.Objects;
+
 /**
  * Implement a list that is based on array. This resizable data structure
  * is based on an array and is a simplified version of {@link java.util.ArrayList}.
  */
 public class ArrayList<T> implements List<T> {
 
+    private Object[] elements;
+
     @Override
     public void add(T element) {
-
+        if (elements.length == size()) {
+            Object[] newElements = new Object[size() * 2];
+            System.arraycopy(element, 0, newElements, 0, size());
+            elements = newElements;
+        }
+        elements[size()] = element;
     }
 
     @Override
     public void add(int index, T element) {
-
+        if (elements.length == size()) {
+            Object[] newElements = new Object[size() * 2];
+            System.arraycopy(element, 0, newElements, 0, size());
+            elements = newElements;
+        }
+        elements[index] = element;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 ||index >= elements.length) {
+            throw new IndexOutOfBoundsException("Input correct index!");
+        }
+        return (T) elements[index];
     }
 
     @Override
     public T get(T element) {
-        return null;
+        if (element == null) {
+            throw new IllegalArgumentException("Input correct data!");
+        }
+
+        int index = -1;
+
+        for (int i = 0; i < elements.length; i++) {
+            if (elements[i].equals(element)) {
+                index = i;
+            }
+        }
+
+        if (index == -1) {
+            return null;
+        }
+        return (T) elements[index];
     }
 
     @Override
     public T getFirst() {
-        return null;
+        if (isEmpty()) {
+            return null;
+        }
+        return get(0);
     }
 
     @Override
     public T getLast() {
-        return null;
+        if (isEmpty()) {
+            return null;
+        }
+        return get(size() - 1);
     }
 
     @Override
     public void set(int index, T element) {
-
+        if ( index < 0 ||index >= elements.length) {
+            throw new IndexOutOfBoundsException("Input correct index!");
+        }
+        elements[index] = element;
     }
 
     @Override
-    public boolean remove(int index) {
-        return false;
+    public T remove(int index) {
+        if ( index < 0 ||index >= elements.length) {
+            throw new IndexOutOfBoundsException("Input correct index!");
+        }
+        T removedEl = (T) elements[index];
+
+        for (int i = 0; i < elements.length; i++) {
+            elements[i] = elements[i + 1];
+        }
+        elements[size() - 1] = null;
```

Comment:
Can be replaced with `System.arraycopy()`. Use it here instead of looping manually. The benefit is better performance of your collection.